CMakeLists.txt

```diff
@@ -780,6 +790,15 @@ add_library (seastar
   src/websocket/server.cc
   )
 
+# Shared library soname  
+set_target_properties(seastar PROPERTIES
+  VERSION ${PROJECT_VERSION}.${Seastar_API_LEVEL}.${Seastar_RELEASE_DATE}
```

Comment:
In semantic versioning (semver), a version number consists of three components: MAJOR.MINOR.PATCH. The major version number changes least frequently and indicates incompatible API changes. The minor version is incremented more often for backward-compatible feature additions, while the patch version is used for backward-compatible bug fixes.

The `Seastar_API_LEVEL` changes less frequently than the ABI version, as it is only incremented for significant API changes rather than ABI-breaking modifications.

`PROJECT_VERSION` is also composed of multiple component. see https://cmake.org/cmake/help/latest/variable/PROJECT_VERSION.html. 

Based on the discussion above, I suggest:

- `PROJECT_VERSION` should be treated as a versioning string with multiple components (major.minor.patch) rather than a single number.
- Although `Seastar_API_LEVEL` is crucial for API versioning, it may not belong in the libraries' version numbers since we make ABI changes within the same `Seastar_API_LEVEL`.
- Including the release date in the version number makes sense given our flexible release schedule.